protocols/imap/src/test/java/org/apache/james/imap/decode/parser/GetAnnotationCommandParserTest.java

```diff
@@ -102,142 +106,224 @@ public void decodeMessageShouldReturnRequestWhenCommandHasMultiKeys() throws Dec
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, outputStream);
+
+        parser.decode(command, lineReader, TAG, session);
+    }
+
+    @Test(expected = DecodingException.class)
+    public void decodeMessageShouldReturnRequestWhenCommandHasDepthOptionAndValueButNotInParenthesis() throws DecodingException {
+        InputStream inputStream = new ByteArrayInputStream((INBOX + " DEPTH 1 (MAXSIZE 1024) (/shared/comment /private/comment) \n").getBytes());
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, outputStream);
+
+        parser.decode(command, lineReader, TAG, session);
     }
 
     @Test
     public void decodeMessageShouldReturnRequestWithZeroDepthOption() throws DecodingException {
         InputStream inputStream = new ByteArrayInputStream((INBOX + " (DEPTH 0) (MAXSIZE 1024) (/shared/comment /private/comment) \n").getBytes());
-        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, outputStream);
 
         GetAnnotationRequest request = (GetAnnotationRequest)parser.decode(command, lineReader, TAG, null);
 
-        assertThat(request.getDepthValue()).isEqualTo(Depth.ZERO);
+        assertThat(request.getDepth()).isEqualTo(Depth.ZERO);
     }
 
     @Test
     public void decodeMessageShouldReturnRequestWithOneDepthOption() throws DecodingException {
         InputStream inputStream = new ByteArrayInputStream((INBOX + " (DEPTH 1) (MAXSIZE 1024) (/shared/comment /private/comment) \n").getBytes());
-        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, outputStream);
 
-        GetAnnotationRequest request = (GetAnnotationRequest)parser.decode(command, lineReader, TAG, null);
+        GetAnnotationRequest request = (GetAnnotationRequest)parser.decode(command, lineReader, TAG, session);
 
-        assertThat(request.getDepthValue()).isEqualTo(Depth.ONE);
+        assertThat(request.getDepth()).isEqualTo(Depth.ONE);
     }
 
     @Test
     public void decodeMessageShouldReturnRequestWhenCommandHasOptionsInAnyOrder() throws DecodingException {
         InputStream inputStream = new ByteArrayInputStream((INBOX + " (MAXSIZE 1024) (DEPTH 1) (/shared/comment /private/comment) \n").getBytes());
-        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, null);
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, outputStream);
 
-        GetAnnotationRequest request = (GetAnnotationRequest)parser.decode(command, lineReader, TAG, null);
+        GetAnnotationRequest request = (GetAnnotationRequest)parser.decode(command, lineReader, TAG, session);
```

Comment:
Yes, we can.
But IMO, It does not need it cause each test functions should be tested for each purpose that it assert, not for all.
But anyway, I follow the rule.